storage/types.go

```diff
@@ -282,20 +282,59 @@ func (n *NodeID) CoordString() string {
 	return fmt.Sprintf("[d:%d, i:%d]", d, i>>d)
 }
 
+// Copy returns a duplicate of NodeID
+func (n *NodeID) Copy() *NodeID {
+	p := make([]byte, len(n.Path))
+	copy(p, n.Path)
+	return &NodeID{
+		Path:          p,
+		PrefixLenBits: n.PrefixLenBits,
+	}
+}
+
+// FlipRightBit flips the ith bit from LSB
+func (n *NodeID) FlipRightBit(i int) *NodeID {
+	n.SetBit(i, n.Bit(i)^1)
+	return n
+}
+
+// leftmask contains bitmasks indexed such that the left x bits are set. It is
+// indexed by byte position from 0-7 0 is special cased to 0xFF since 8 mod 8
+// is 0. leftmask is only used to mask the last byte.
+var leftmask = [8]byte{0xFF, 0x80, 0xC0, 0xE0, 0xF0, 0xF8, 0xFC, 0xFE}
+
+// MaskLeft returns NodeID with only the left n bits set
+func (n *NodeID) MaskLeft(depth int) *NodeID {
+	r := make([]byte, len(n.Path))
+	if depth > 0 {
+		// Copy the first depthBytes.
+		depthBytes := bytesForBits(depth)
+		copy(r, n.Path[:depthBytes])
+		// Mask off unwanted bits in the last byte.
+		r[depthBytes-1] = r[depthBytes-1] & leftmask[depth%8]
+	}
+	if depth < n.PrefixLenBits {
+		n.PrefixLenBits = depth
+	}
+	n.Path = r
+	return n
+}
+
+// Neighbor sets this node to be it's own neighbor.
```

Comment:
This is doing exactly the same thing as the previous version of `Sibling`. The steps have just been broken up for clarity. 

`Neighbor` is doing the following snippet from the previous: 

```
bi = n.PathLenBits() - n.PrefixLenBits + i ==
height = n.PathLenBits() - (n.PrefixLenBits - i)
r[i].SetBit(bi, n.Bit(bi)^1)
```

Mask Left is doing this snippet
```
r[i].PrefixLenBits =  n.PrefixLenBits - i
```